include/engine/guidance/step_maneuver.hpp

```diff
@@ -11,12 +13,21 @@ namespace engine
 namespace guidance
 {
 
+enum class WaypointType : std::uint8_t
+{
+    None,
+    Arrive,
+    Depart,
+    Waypoint
```

Comment:
This does not exist anymore because of the leg split. I would expect the instructions for a leg to be:
`Depart, [...], Arrive`. I don't think we need a `Waypoint` type - for instructions our biggest unit is a leg, hence there are no via.